components/connector/internal/revocation/repository.go

```diff
@@ -3,31 +3,37 @@ package revocation
 import (
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
 	"k8s.io/client-go/util/retry"
 )
 
 //go:generate mockery -name=Manager
 type Manager interface {
 	Get(name string, options metav1.GetOptions) (*v1.ConfigMap, error)
 	Update(configmap *v1.ConfigMap) (*v1.ConfigMap, error)
+	Watch(opts metav1.ListOptions) (watch.Interface, error)
 }
 
 //go:generate mockery -name=RevocationListRepository
 type RevocationListRepository interface {
 	Insert(hash string) error
-	Contains(hash string) (bool, error)
+	Contains(hash string) bool
 }
 
 type revocationListRepository struct {
-	configListManager Manager
-	configMapName     string
+	configListManager   Manager
+	configMapName       string
+	revocationListCache Cache
 }
 
-func NewRepository(configListManager Manager, configMapName string) RevocationListRepository {
-	return &revocationListRepository{
-		configListManager: configListManager,
-		configMapName:     configMapName,
+func NewRepository(configListManager Manager, revocationListCache Cache, configMapName string) RevocationListRepository {
+	obj := &revocationListRepository{
+		configListManager:   configListManager,
+		configMapName:       configMapName,
+		revocationListCache: revocationListCache,
 	}
+
+	return obj
```

Comment:
Is there any specific reason to create an object and then return it instead of just returning a newly constructed object? Just like before.